### PR TITLE
#1768 - Transfer_suffix required for directory strikes

### PR DESCRIPTION
**Pull request by Fizz11**
##### Checklist

- [x] `manage.py test` passes
- [x] tests are included
- [x] documentation is changed or added

### Affected app(s)

ingest

### Description of change

made transfer_suffix optional for directory strikes.

### CODE REVIEW

**Comment by cshamis** [comment]
Transfer suffix may be optional, but without it, strike will have NO WAY of knowing if a file is done transferring or not.

The transfer suffix should only be left undefined if the user KNOWS FOR CERTAIN that transfers are atomic.

We might want to warn them of that.

File: scale/docs/rest/v6/strike.rst [hunk 1]
```diff
@@ -521,7 +521,7 @@ A strike configuration JSON describes a set of configuration settings that affec
 +----------------------------+----------------+----------+--------------------------------------------------------------------+
 | .type                      | String         | Required | The type of the monitor. Must be either 'dir-watcher' or 's3'      |
 +----------------------------+----------------+----------+--------------------------------------------------------------------+
-| .transfer_suffix           | String         | Required | (dir-watcher)Defines a suffix that is used on the file names to    |
+| .transfer_suffix           | String         | Optional | (dir-watcher)Defines a suffix that is used on the file names to    |
```
Comment (emimaesmith) [inline scale/docs/rest/v6/strike.rst]: I would document here that if it's not provided, then a default of '_tmp' will be used.

File: scale/ingest/strike/monitors/dir_monitor.py [hunk 2]
```diff
@@ -78,13 +81,11 @@ def stop(self):
...
         """See :meth:`ingest.strike.monitors.monitor.Monitor.validate_configuration`
         """
-
-        if 'transfer_suffix' not in configuration:
-            raise InvalidMonitorConfiguration('transfer_suffix is required for dir-watcher monitor')
-        if not isinstance(configuration['transfer_suffix'], basestring):
-            raise InvalidMonitorConfiguration('transfer_suffix must be a string')
-        if not configuration['transfer_suffix']:
-            raise InvalidMonitorConfiguration('transfer_suffix must be a non-empty string')
+        if 'transfer_suffix' in configuration:
```
Comment (emimaesmith) [inline scale/ingest/strike/monitors/dir_monitor.py]: Is there any way to return a warning if transfer_suffix isn't defined? Something that will return saying something like "No transfer suffix defined, '_tmp' will be used"?

File: scale/ingest/test/strike/monitors/test_dir_monitor.py [hunk 3]
```diff
@@ -18,7 +18,7 @@ def test_validate_configuration_missing_transfer_suffix(self):
         config = {
             'type': 'dir-watcher'
         }
-        self.assertRaises(InvalidMonitorConfiguration, DirWatcherMonitor().validate_configuration, config)
+        DirWatcherMonitor().validate_configuration(config)
```
Comment (emimaesmith) [inline scale/ingest/test/strike/monitors/test_dir_monitor.py]: If we don't care about transfer_suffix being there or not, might as well just remove this test.

**Pull request merged**